### PR TITLE
Added test cases

### PR DESCRIPTION
**Pull request by Layman806**
I'm sorry, @debck . I accidentally pushed changes to 'master' without creating a new branch first. Then I reverted it and created a new branch 'test_cases' from master. This has changes that you can directly merge, although GitHub says it cannot automatically merge.

### CODE REVIEW

**Comment by Layman806** [comment]
PR Quality Review Failed. LOL. I use vim, and automatically use TABs instead of spaces, so it flags every line!
Do I need to change stuff?

File: bin/app.js [hunk 1]
```diff
@@ -1,22 +1,4 @@
...
-  // console.log(input);
-  const { filename, extension } = input;
-  // console.log(filename);
-  create.create(filename, extension);
-
-  console.log(chalk.blue('File created successfully') + chalk.red('!!!'));
-};
-
-main();
+sub.main();
```
Comment (debck) [inline bin/app.js]: I dont think the new file `sub.js` is necessary. you can declare the same in the `app.js`

Comment (Layman806) [reply bin/app.js]: Yeah, I don't remember why I did that. I put it back in app.js with the latest commit.

**Comment by Layman806** [comment]
AHA! Fixed the TravisCI Build as well!
(#11) solved.

**Comment by debck** [comment]
Can u squash the commits. @Layman806

File: bin/app.js [hunk 2]
```diff
@@ -1,14 +1,12 @@
...
-const shell = require('shelljs');
 const inquirer = require('inquirer');
 const chalk = require('chalk');
 const header = require('../lib/header');
 const question = require('../lib/question');
 const create = require('../lib/create');
 const ifdir = require('../lib/ifdir');
 
-const main = async () => {
+exports.main = async () => {
```
Comment (debck) [inline bin/app.js]: make it `const main `

File: bin/app.js [hunk 3]
```diff
@@ -20,15 +18,18 @@ const main = async () => {
...
-	}
-};
 
-main();
+		return 0;
+	} else {
+		// Invalid response in isdir
+		return 2;
+	}
+}
```
Comment (debck) [inline bin/app.js]: after the `}` ends write `main();` at last

**Comment by debck** [comment]
Just make the changes mentioned and rest looks LGTM

File: bin/app.js (hunk 3 above)
Comment (debck) [inline bin/app.js]: Just after the `}` write `main()`

File: bin/app.js (hunk 2 above)
Comment (debck) [inline bin/app.js]: change this to `const main = exports.main = async () =>`

**Comment by debck** [comment]
I think there is some issue in Travis setup. The tests were passed locally.
LGTM. @Layman806  Just squash the commits and I will merge this. 
travis issue will be solved later

**Comment by debck** [comment]
We can solve this issue in another PR

**Comment by Layman806** [comment]
> I think there is some issue in the test you wrote , jest is not terminating after the test passes . However the tests were passed locally.
> LGTM. @Layman806 Just squash the commits and I will merge this.

Remember when I separated main() from app and into a separate file called sub.js?
That actually tackled this build testing issue. I worked around by not having to use 'await' in the top level code, which is not allowed. 
I reverted it back so that the app.js now only has one statement to run.

**Comment by Layman806** [comment]
> I think there is some issue in the test you wrote , jest is not terminating after the test passes . However the tests were passed locally.
> LGTM. @Layman806 Just squash the commits and I will merge this.

I'll squash it first thing tomorrow.

**Comment by debck** [comment]
Thanks @Layman806 for fixing this.
That was a mistake from my side I should have taken some time to understand your point .
Thanks :tada:

**Comment by debck** [comment]
> I'll squash it first thing tomorrow.
I did that from my side . No need . You can continue your contribution. :+1:

**Pull request merged**